DicomRtImportExport/Logic/vtkSlicerDicomRtReader.cxx

```diff
@@ -128,6 +148,8 @@ class vtkSlicerDicomRtReader::vtkInternal
     double BeamLimitingDeviceAngle;
     /// Jaw positions: X and Y positions with isocenter as origin (e.g. {{-50,50}{-50,50}} )
     double LeafJawPositions[2][2];
+    /// MLC positions: Raw DICOM values [0] - MLCX, [1] - MLCY
+    std::vector<double> LeafJawPositionsMLC[2];

```

Comment:
You added the class BeamLimitingDeviceEntry, which is quite nice, but then you add this array of vectors in the beam, which seems a bit disconnected. I'd just like to understand...

So the BeamLimitingDeviceEntry is the general description of one MLC array, and these two vectors are the actual positions of the leaves for the beam, correct? How do we know what corresponds to what? Simply assume that the first leaf position vector corresponds to the first MLC device?

Reply:
Correct. BeamLimitingDeviceEntry is a description of MLC (or any beam limiting device such as symmetric or asymmetric jaws).

DICOM standard describes two kinds of multi-leaf collimators: "MLCX" - leaves moves along X-axis, "MCLY" - leaves moves along Y-axis.

LeafJawPositionsMLC[0] - leaf position vector for MLCX
LeafJawPositionsMLC[1] - leaf position vector for MLCY

Leaf position vector stores data as it had been written with in DICOM file. (60 leaf pairs = 120 leaf positions)

Reply:
Thanks for the explanation!

Reply:
Could you add some of this text to the code as comment? Adding meaningful comments facilitates maintenance and future improvements a lot! Thanks